### PR TITLE
Added backticks to spath keys and enabled the tests

### PR DESCRIPTION
**Pull request by Abigael-JT**
changes made:
- Added backticks to spath keys to escape dots in the path 
- Enabled eval spath test 
- changed expected column arrangement in aggregateAfterHdfsLoadTest

Related ticket
* #330

### CODE REVIEW

File: src/test/java/com/teragrep/pth10/evalTest.java [hunk 1]
```diff
@@ -2609,7 +2609,7 @@ public void parseEvalSpathJSONCatalystTest() {
     // Test spath() with XML
     // //person[age=30]/name/text()
     // FIXME broken due to spath udf changes
-    @Disabled
+    //@Disabled
```
Comment (elliVM) [inline src/test/java/com/teragrep/pth10/evalTest.java:2612]: Could this line be removed instead of commenting out?

**Review by elliVM — CHANGES_REQUESTED** [review]
Applying of backticks could be done by an object that wraps the String with methods for formatted and original String values. Alternatively formatted Strings could be created at the start of the method with names like `stringWithBackticks` to clear the code a bit. This solution is also fine those changes are up to you.

### Check this please
Seems like some code format was changed. Please run  `mvn spotless:apply` to check for changes so it doesn't pop up on other branches. Otherwise looks good

**Comment by Abigael-JT** [comment]
> Applying of backticks could be done by an object that wraps the String with methods for formatted and original String values. Alternatively formatted Strings could be created at the start of the method with names like `stringWithBackticks` to clear the code a bit. This solution is also fine those changes are up to you.
> ### Check this please
> 
> Seems like some code format was changed. Please run `mvn spotless:apply` to check for changes so it doesn't pop up on other branches. Otherwise looks good

Removed commented out code and added objects to escape and unescape backticks in spath keys, also added tests for those new objects. We also changed one of the spath test  to only filter out the duplicates and now there is only one assertion.

File: src/main/java/com/teragrep/pth10/steps/spath/SpathEscapedKey.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package com.teragrep.pth10.steps.spath;import java.io.Serializable;
```
Comment (elliVM) [inline src/main/java/com/teragrep/pth10/steps/spath/SpathEscapedKey.java:1]: Add license. `mvn spotless:apply` should create missing licenses

File: src/main/java/com/teragrep/pth10/steps/spath/SpathUnescapedKey.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package com.teragrep.pth10.steps.spath;
```
Comment (elliVM) [inline src/main/java/com/teragrep/pth10/steps/spath/SpathUnescapedKey.java:1]: Missing license

File: src/test/java/com/teragrep/pth10/SpathEscapedKeyTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
+package com.teragrep.pth10;
```
Comment (elliVM) [inline src/test/java/com/teragrep/pth10/SpathEscapedKeyTest.java:1]: Missing license

File: src/test/java/com/teragrep/pth10/SpathUnscapedKeyTest.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
+package com.teragrep.pth10;
```
Comment (elliVM) [inline src/test/java/com/teragrep/pth10/SpathUnscapedKeyTest.java:1]: Missing license

**Review by elliVM — CHANGES_REQUESTED** [review]
Changes look good to me just missing licenses and to apply spotless

**Review by elliVM — APPROVED** [review]
LGTM

**Comment by eemhu** [comment]
Added missing SpathEscapedKey in evalStatement, rebased onto main and re-enabled disabled spath tests

**Review by elliVM — APPROVED** [review]

File: src/main/java/com/teragrep/pth10/steps/spath/SpathEscapedKey.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.pth10.steps.spath;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+public final class SpathEscapedKey implements Serializable {
```
[the remark this answers is not in the document]
Comment (kortemik) [reply src/main/java/com/teragrep/pth10/steps/spath/SpathEscapedKey.java:51]: SpathEscapedKey should have a method unescape that returns SpathKey

**Review by kortemik — CHANGES_REQUESTED** [review]

File: src/main/java/com/teragrep/pth10/steps/spath/SpathEscapedKey.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (Abigael-JT) [reply src/main/java/com/teragrep/pth10/steps/spath/SpathEscapedKey.java:51]: This has been fixed

File: src/main/java/com/teragrep/pth10/steps/spath/SpathEscapedKey.java [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+    private static final long serialVersionUID = 1L;
+    private final String key;
+
+    public SpathEscapedKey(final String key) {
+        this.key = key;
+    }
+
+    public SpathKey unescaped() {
```
Comment (kortemik) [inline src/main/java/com/teragrep/pth10/steps/spath/SpathEscapedKey.java:60]: I was thinking this during the day ans thought that is this actually an escaped thing or is it rather a backtick quoted string and usual unquote functionality should.be used? @eemhu Any comments as you did previous quotation fixes

Comment (eemhu) [reply src/main/java/com/teragrep/pth10/steps/spath/SpathEscapedKey.java:60]: I suppose strictly speaking it is more of a backtick quotation but logically it is used to escape `.` (dots), so they can be used in the column names.

Comment (kortemik) [reply src/main/java/com/teragrep/pth10/steps/spath/SpathEscapedKey.java:60]: I think we should make the object re-usable for such purposes and use the generic quotation object that allows specifying the quotation character. this way it would be able to handle 
```
' " `
```
 and possibly other characters as well. current substring way does not handle properly escapes in a string. i.e. 
```
`foo\`bar`
```
 should be unquoted as 
```
foo`bar
```

Comment (eemhu) [reply src/main/java/com/teragrep/pth10/steps/spath/SpathEscapedKey.java]: Added backtick support for existing UnquotedText object and also added QuotedText object.

File: src/main/java/com/teragrep/pth10/ast/UnquotedText.java [hunk 8]
```diff
@@ -53,7 +54,7 @@
  */
 public class UnquotedText implements Text {
```
Comment (kortemik) [inline src/main/java/com/teragrep/pth10/ast/UnquotedText.java:55]: i think this should take the quote character as an argument instead of using regex to match it?

**Review by kortemik — CHANGES_REQUESTED** [review]
i would add a constructor parameter instead of using regex because regex is not very readable.

otherwise looks very clean and usable

File: src/main/java/com/teragrep/pth10/ast/UnquotedText.java (hunk 8 above)
Comment (eemhu) [reply src/main/java/com/teragrep/pth10/ast/UnquotedText.java:55]: Would it be okay if no quote character is provided it would remove all, and if it is provided it only removes the specific one? I think currently UnquotedText is used to remove all on purpose

**Review by kortemik — APPROVED** [review]

**Comment by kortemik** [comment]
good code: merging!
